### PR TITLE
Redux/actions reducers store

### PR DESCRIPTION
**Pull request by EstebanIndiveri**
## Summary

Added a Redux, react-redux redux-thunk, create a store, actions and reducers, ready to use in next pull request.

## Trello Card

[Redux Trello Card](https://trello.com/c/vXZsXBNq/25-n-91-redux-i-actions-reducers-store)

### CODE REVIEW

File: wbooks/ReactotronConfig.js [hunk 1]
```diff
@@ -1,4 +1,10 @@
...
 import Reactotron from 'reactotron-react-native';
 import AsyncStorage from '@react-native-async-storage/async-storage';
 
-Reactotron.setAsyncStorageHandler(AsyncStorage).configure().useReactNative().connect();
+const reactotron = Reactotron.configure({ name: 'React Native Demo' })
+  .setAsyncStorageHandler(AsyncStorage)
+  .use(reactotronRedux())
+  .useReactNative({})
+  .connect();
+export default reactotron;
```
Comment (andreszuluaga-wolox) [inline wbooks/ReactotronConfig.js]: this file should be inside config folder

File: wbooks/src/app/index.tsx [hunk 2]
```diff
@@ -2,14 +2,19 @@ import React, { FC } from 'react';
 import { StatusBar } from 'react-native';
 import { NavigationContainer } from '@react-navigation/native';
 import Navigator from '@navigator';
+import { Provider } from 'react-redux';
+
+import { store } from '../redux/store';
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/index.tsx]: the folder redux should have alias

File: wbooks/src/redux/book/actions.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { BOOKS_MOCK } from '@constants/mockBooks';
+import { Dispatch } from 'redux';
+
+import { BookInterface, BookType } from '../../interfaces';
```
Comment (andreszuluaga-wolox) [inline wbooks/src/redux/book/actions.ts]: fix the error and make push again

File: wbooks/src/redux/store.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { createStore } from 'redux';
+
+import Reactotron from '../../ReactotronConfig';
```
Comment (andreszuluaga-wolox) [inline wbooks/src/redux/store.tsx]: use alias

File: wbooks/src/interfaces/index.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  id: number;
+  author: string;
+  title: string;
+  genre: string;
+  publisher: string;
+  year: string;
+  imageUrl: string | null;
+}
+export enum BookType {
+  BOOK_LIST_REQUEST = 'BOOK_LIST_REQUEST',
```
Comment (mattgle) [inline wbooks/src/interfaces/index.ts]:
```suggestion
  BOOK_LIST = 'BOOK_LIST',
```

File: wbooks/src/interfaces/index.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  BOOK_LIST_SUCCESS = 'BOOK_LIST_SUCCESS',
+  BOOK_LIST_FAIL = 'BOOK_LIST_FAIL'
```
Comment (mattgle) [inline wbooks/src/interfaces/index.ts]:
```suggestion
  BOOK_LIST_ERROR = 'BOOK_LIST_ERROR'
```

File: wbooks/src/redux/book/reducer.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+interface BookState {
+  loading: boolean;
+  books: BookInterface[];
+  error?: boolean;
+}
+
+const initalState: BookState = {
+  books: [],
+  loading: false,
```
Comment (mattgle) [inline wbooks/src/redux/book/reducer.ts]:
```suggestion
  booksLoading: false,
```

File: wbooks/src/redux/book/reducer.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  error: false
```
Comment (mattgle) [inline wbooks/src/redux/book/reducer.ts]: booksError

File: wbooks/tsconfig.json [hunk 9]
```diff
@@ -63,10 +63,11 @@
       "@screens/*":["src/app/screens/*"],
       "@config/*":["src/config/*"],
       "@hooks/*":["src/app/hooks/*"],
-      "@interfaces":["src/interfaces/*"],
```
Comment (mattgle) [inline wbooks/tsconfig.json:66]: this was ok

File: wbooks/src/redux/book/actions.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  const fakeBooksFetch: BookInterface[] = [];
+  try {
+    await BOOKS_MOCK.forEach(book => {
+      fakeBooksFetch.push(book);
+    });
+    return { status: 'ok', data: fakeBooksFetch };
+  } catch (error) {
+    return { message: 'Not today' };
+  }
+};
```
Comment (mattgle) [inline wbooks/src/redux/book/actions.ts]: I think this is not needed, you can just make a promise and resolve it.

File: wbooks/src/redux/book/actions.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+      dispatch({
+        type: BookType.BOOK_LIST_FAIL,
+        payload:
+          error.response && error.response.message
+            ? error.response.message
+            : "We can't load library now, try again in some minutes. Thank you"
+      });
+    }
+  }
+};
```
Comment (mattgle) [inline wbooks/src/redux/book/actions.ts]: i think this should be a normal action request, read redux documentation again and try to give it another try.

**Review by mattgle — CHANGES_REQUESTED** [review]

File: wbooks/src/app/index.tsx [hunk 12]
```diff
@@ -2,14 +2,18 @@ import React, { FC } from 'react';
...
     <>
-      <StatusBar backgroundColor="transparent" barStyle="dark-content" />
-      <NavigationContainer>
-        <Navigator />
-      </NavigationContainer>
+      <Provider store={store}>
+        <StatusBar backgroundColor="transparent" barStyle="dark-content" />
+        <NavigationContainer>
+          <Navigator />
+        </NavigationContainer>
```
Comment (mattgle) [inline wbooks/src/app/index.tsx]: i think navigationContainer can be inside Navigator, do the change if possible

File: wbooks/src/redux/book/reducer.ts [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import { BOOKS_MOCK } from '@constants/mockBooks';
+import { BookInterface, BookType, BookActionsTypes } from 'src/interfaces/book';
```
Comment (mattgle) [inline wbooks/src/redux/book/reducer.ts]:
```suggestion
import { BookInterface, BookType, BookActionsTypes } from '@interfaces/book';
```

**Review by mattgle — CHANGES_REQUESTED** [review]

**Review by mattgle — DISMISSED** [review]

**Review by mattgle — APPROVED** [review]

**Pull request merged**